Creados los árboles que spawnean armas

Pj6595 opened the pull request:
### Árboles de spawns

Se han creado los árboles que spawnean armas. La gracia es que va apareciendo el arma en cuestión y se va haciendo grande hasta que pasa el tiempo de spawn y se puede utilizar.

**CLASE SPAWNTREE**
Los árboles están en la capa UninteractableObject. La posición desde la que spawnean las cosas es un poco provisional porque no sé exactamente cómo va a ser el árbol/maceta/loquesea que vaya a ser esto al final, la ajustaré cuando se sepa.

**CLASE WEAPONPOOL**
Para que los árboles puedan spawnear armas de cualquier pool de armas, se ha creado la clase WeaponPool de la que heredarán todas las pools de objetos que vayan a spawnear de árboles. Básicamente es un interfaz que permite que por polimorfismo podamos pasarle al árbol cualquier pool de armas y funcione bien sin necesidad de más movidas chungas, toda la funcionalidad sigue estando en la clase específica de cada pool.

También se han  modificado los tilemaps (y la clase pertinente) para que se puedan añadir los árboles de spawn desde tiled.

mmur07 committed:
el tomate "funciona" pero falta mucho

necesita ser gestionado por pools, ser controlado por las hands del player, y funcionar por la arquitectura de Weapon, además de hacer bien lo del timer.

EstebenR committed:
Creados ActionableWeapon y MeleeWeapon

EstebenR committed:
Compila! c:

EstebenR committed:
Las armas a melee ya se pueden usar

EstebenR committed:
Limpieza

EstebenR committed:
Merge branch 'master' into 172410728-Melee-Weapons

EstebenR committed:
Miguel was right al along

EstebenR committed:
Update SlipperWeapon.cpp

EstebenR committed:
Okrr

Co-Authored-By: Jorge Moreno Martínez <jorgmo02@ucm.es>

EstebenR committed:
Vaya fallo mas tonto xd

mmur07 committed:
El tomate explota usando la animación y ya usa cooldowns bien

mmur07 committed:
el tomate suelta una anilla al activarse

EstebenR committed:
Añadidas particulas maximas

mmur07 committed:
establecido numero max. de partículas al activar el tomate

mmur07 committed:
Merge branch '172410728-Melee-Weapons' into 171293110-TomateGranada

mmur07 committed:
TENEMOS TOMATADA. Falta ajustar lo del spawn de las partículas una vez esté apañado lo de las armas y todo el tema de sprites al recoger el arma

EstebenR committed:
Some pre-destroy cleanup

EstebenR committed:
Ya se puede hacer daño

Yes I would like to die, that'd be very kind

EstebenR committed:
Merge branch 'master' into 172410728-Melee-Weapons

EstebenR committed:
ahora si que matan y yo me quiero morir

EstebenR committed:
Cambiado nombre de component slipperWeapon

Do you ever look at someone and wonder... what is going on inside their heads?

EstebenR committed:
Hola la mancuerna no hace daño porque no tiene daño metido

AdriAlvarez00 committed:
Fixup monedas

AdriAlvarez00 committed:
Merge remote-tracking branch 'origin/172410728-Melee-Weapons' into 172410728-Melee-Weapons

mmur07 committed:
se guardan las hands del player al ser recogidas pero necesito el collider de algo para spawnear la partícula

mmur07 committed:
Merge branch '172410728-Melee-Weapons' into 171293110-TomateGranada

mmur07 committed:
ESTAMOS MÁS CERCA DE LA VERDAD TOMATIL

mmur07 committed:
los tomates ya se spawnean por pools pero hace falta adaptarlo a timeGameMode

mmur07 committed:
tomate ya funcional. quiero experimentar un poco más con la animación

mmur07 committed:
olvidemonos todos de la segunda textura de la explosión del tomate

mmur07 committed:
arreglos finales

title

mmur07 committed:
Merge branch 'master' into 171293110-TomateGranada

mmur07 committed:
La fuerza de la explosión ahora es una constante

Pj6595 committed:
Arquitectura básica del SpawnTree terminada

Queda añadir los tomates granada

Pj6595 committed:
Merge branch '171293110-TomateGranada' into spawn-trees

Pj6595 committed:
Funciona (más o menos)

Me queda hacer otro refactor de pools, ajustar las posiciones de spawn y descurbrir por qué los tomates colisionan con los roombas

Pj6595 committed:
Pools refactorizadas + otros ajustes

mmur07 committed:
cambiado el tamaño de la pool de tomates a constante

mmur07 committed:
Update src/SourceFiles/TomatoPool.h

oko thanks

Co-Authored-By: Jorge Moreno Martínez <jorgmo02@ucm.es>

mmur07 committed:
ya no colisiona con el router al ser lanzado

mmur07 committed:
cambio de nombres de variables y ahora llama a OnPlayerDead()

mmur07 committed:
cambio de nombre de clase de TomatoLogic -> TomatoWeapon

Pj6595 committed:
Merge branch '171293110-TomateGranada' into spawn-trees

AndresRuizB reviewed (commented):
Sigue así pejota ;D

Pj6595 committed:
Creados los bananaTrees

Pj6595 committed:
Merge branch 'master' into spawn-trees

Pj6595 committed:
Terminados árboles de tomates y plátanos + incluidos en el tilemap

JuanmaSalva reviewed (approved):
el platano esta roto :(

EstebenR commented on src/SourceFiles/TimeGameMode.h:
```diff
@@ -24,6 +24,8 @@ class TimeGameMode : public AbstractTimedGameMode
 
 	void renderKillMarker();
 
+	TomatoPool tomatoPool_;
```
Esto deberia estar aqui?

EstebenR commented on src/SourceFiles/CapitalismGameMode.cpp:
```diff
@@ -1,5 +1,8 @@
 #include "CapitalismGameMode.h"
 #include "FireBallGenerator.h"
+#include "ObjectFactory.h"
+#include "TomatoPool.h"
+#include "BananaPool.h"
```
Los pools hace falta incluirlos? Por que en el resto de gamemodes no?

EstebenR reviewed (changes requested):
Los archivos de TomatoTree estan comentados enteros??

En StocksGameMode sigue estando el need to delete, linea 17

JuanmaSalva reviewed (approved):
ahora si, antes lo habia aceptado aun q habia algo roto por q soy retrasado y me habia equivocado de boton

Pj6595 committed:
Fix definitivo de los árboles spawners

Ya funcionan los plátanos que salen de árboles.

Juanma vive en el futuro y ya ha aceptado este commit.

Pj6595 committed:
Merge branch 'master' into spawn-trees

Adrian-de-Lucas-Gomez reviewed (approved):
Nice

EstebenR reviewed (changes requested):
Al disparar el platano no sale proyectil
